containers/daap-delete-table-for-data-product/src/var/task/delete_table.py

```diff
@@ -0,0 +1,173 @@
+import os
+from http import HTTPStatus
+
+import boto3
+from botocore.exceptions import ClientError
+from data_platform_api_responses import format_error_response
+from data_platform_logging import DataPlatformLogger
+from data_platform_paths import DataProductElement, get_metadata_bucket
+from data_product_metadata import DataProductMetadata, DataProductSchema
+
+logger = DataPlatformLogger(
+    extra={
+        "image_version": os.getenv("VERSION", "unknown"),
+        "base_image_version": os.getenv("BASE_VERSION", "unknown"),
+    }
+)
+
+s3_client = boto3.client("s3")
+s3_resource = boto3.resource("s3")
+glue_client = boto3.client("glue")
+athena_client = boto3.client("athena")
+
+
+def get_all_versions(data_product_name: str) -> list[str]:
+    """Get the latest version of a data product, else return v1.0"""
+    metadata_bucket = get_metadata_bucket()
+    metadata_versions = s3_client.list_objects_v2(
+        Bucket=metadata_bucket, Prefix=data_product_name + "/"
+    )
+    # This is the case in which the data product is new.
+    if not metadata_versions.get("Contents"):
+        return ["v1.0"]
+
+    versions = [
+        version["Key"].split("/")[1]
+        for version in metadata_versions["Contents"]
+        if version["Size"] > 0
+    ]
+    versions.sort(key=lambda x: [int(y.replace("v", "")) for y in x.split(".")])
+    print(versions)
+
+    return versions
+
+
+def generate_all_element_version_prefixes(
```

Comment:
we might want to move some of this into the base image, but the logic looks good.

I wish there was a way to just pass a wildcard into the delete function but I don't think there is 😞